### PR TITLE
Fix #609 - Hoodie default page not showing

### LINKED ISSUE #609: Hoodie default page not showing
I am running through the getting started guide and when I run the app without a public folder, I don't see the hoodie default page telling me that I need a `public` folder.

Also, when I navigate to any url e.g localhost:8080/home.html or any, it doesn't return a 404 or something.  Don't know if that's normal

### PR DESCRIPTION
**Pull request by minrwhite**
closes #609

### CODE REVIEW

File: cli/webroot-locator.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = locateWebroot
+var fs = require('fs')
```
Comment (gr2m) [inline cli/webroot-locator.js:3]: 🎨 can you please add an empty line below `module.exports = locateWebroot`

File: test/unit/cli/webroot-locator-test.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+var statsMock = {
+  'isDirectory': simple.spy()
+}
+
+test('webroot locator', function (group) {
+  var webrootLocator = proxyquire('../../../cli/webroot-locator',
+    {fs: fsMock})
+
+  group.afterEach(function (done) {
+    fsMock.existsSync.actions.length = 0
```
Comment (gr2m) [inline test/unit/cli/webroot-locator-test.js]: what is the `.actions.length = 0` for?

File: test/unit/cli/options-test.js [hunk 3]
```diff
@@ -36,6 +36,8 @@ var packageJsonMock = {
   version: '1.0.0'
 }
 
+var mockWebrootLocator = simple.spy(function (arg) { return arg })
```
Comment (gr2m) [inline test/unit/cli/options-test.js]: I didn’t know that you can stub `simple.spy`s, too, but for coherence with other code, can you do `simple.stub()` here and in the other places please?

**Review by gr2m — CHANGES_REQUESTED** [review]
looking great, thanks! Just a few comments

**Comment by minrwhite** [comment]
actions.length = 0 is a way of resetting the stubbed behaviour, as spy.reset() only resets the logged calls. That way if we forget to change the behaviour between tests this should catch it and prevent a false positive.

Currently this fails, as public plugin currently determines the public webroot from the wrong config object path. Correcting this causes other tests to fail; I'll fix these tests as my next course of action.

**Comment by gr2m** [comment]
but you don’t use `actions.length` anywhere else in the code? Am I missing something?

**Comment by minrwhite** [comment]
It's shorthand for emptying an array. I can change this to be a bit more explicit if required?

**Comment by minrwhite** [comment]
Rebased and pushed, ready for review, also pending yay/nay on the `actions.length = 0` array-emptying shorthand.

File: test/unit/cli/webroot-locator-test.js (hunk 2 above)
Comment (minrwhite) [reply test/unit/cli/webroot-locator-test.js]: Comment moved from below: It's shorthand for emptying an array. I can change this to be a bit more explicit if required?

File: test/unit/cli/options-test.js [hunk 4]
```diff
@@ -36,6 +36,8 @@ var packageJsonMock = {
   version: '1.0.0'
 }
 
+var mockWebrootLocator = simple.stub(function (arg) { return arg })
```
Comment (gr2m) [inline test/unit/cli/options-test.js]: you can remove `function (arg) { return arg }`

File: test/unit/cli/webroot-locator-test.js (hunk 2 above)
Comment (gr2m) [reply test/unit/cli/webroot-locator-test.js]: okay I understand what `stub.actions` are now. I didn’t know one can chain `callbackWith`, `returnWith` and `throwWith`, that’s pretty cool! I would think that `stub.reset()` would remove these though, I’ll file an issue on simple-mock.

But please change the `actions.length = 0` to `actions = []`, it’s shorter and more readable

File: test/unit/cli/webroot-locator-test.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  'statSync': simple.stub()
+}
+
+var statsMock = {
+  'isDirectory': simple.stub()
+}
+
+test('webroot locator', function (group) {
+  var webrootLocator = proxyquire('../../../cli/webroot-locator',
+    {fs: fsMock})
```
Comment (gr2m) [inline test/unit/cli/webroot-locator-test.js]: can you please change this to 

``` js
var webrootLocator = proxyquire('../../../cli/webroot-locator', {
  fs: fsMock
})
```

**Review by gr2m — CHANGES_REQUESTED** [review]
okay final comments. I can also fix it up myself if you are busy?

**Comment by minrwhite** [comment]
Actioned review changes, and rebased ready for review.

File: test/unit/cli/options-test.js (hunk 4 above)
Comment (minrwhite) [reply test/unit/cli/options-test.js]: Indeed I can! Done.

File: test/unit/cli/webroot-locator-test.js (hunk 5 above)
Comment (minrwhite) [reply test/unit/cli/webroot-locator-test.js]: Changed. Agreed that that's clearer.

**Review by gr2m — APPROVED** [review]

**Comment by gr2m** [comment]
beautiful work, thanks a ton

**Comment by brunopedroso** [comment]
ooops.. should we check if the `server.paths` key is present? otherwise I'm forced to define it like

```
paths:{}
```

I'm using ember-hoodie...

**Comment by gr2m** [comment]
it gets defaulted here: https://github.com/hoodiehq/hoodie/blob/047a8ff004538e68ccf2a89df33108fec505c3f3/server/index.js#L35-L40

**Comment by brunopedroso** [comment]
yes, but only if I [dont define](https://github.com/hoodiehq/hoodie/blob/047a8ff004538e68ccf2a89df33108fec505c3f3/server/index.js#L26) `db.url`.

We're setting `db.url` here, and our app started to break on [this line](https://github.com/hoodiehq/hoodie/pull/627/files#diff-43557d6060003ff6997c94943ed23a82R11) just after I made the update. I had to define `paths:{}`...

Maybe that line could read like this?

```
var publicFolder = (options.config.paths && options.config.paths.public) || 'public'
```

**Comment by minrwhite** [comment]
FWIW, I'd be tempted to resolve this by moving [this clause](/hoodiehq/hoodie/blob/047a8ff004538e68ccf2a89df33108fec505c3f3/server/index.js#L35) alongside [this one](/hoodiehq/hoodie/blob/047a8ff004538e68ccf2a89df33108fec505c3f3/server/index.js#L19), removing the 'data' key and letting [this](/hoodiehq/hoodie/blob/047a8ff004538e68ccf2a89df33108fec505c3f3/server/index.js#L41) handle the data key.

**Comment by gr2m** [comment]
@brunopedroso ooops that is definitely wrong

The way it’s currently handled is not very elegant. I’ve a lot of work stuff to catch up with and won’t be able to look into this, but if one of you could send a PR that will fix @brunopedroso’s problem, I’m happy to review and merge

**Pull request merged**